File: frontend/scripts/create-pwa-icons.js
#!/usr/bin/env node
/**
 * Create simple PWA icons using Node.js canvas
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Create a simple SVG icon
function createSVGIcon(size, text) {
  const svg = `
<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
  <rect width="${size}" height="${size}" fill="#0f172a"/>
  <text x="50%" y="50%" font-family="Arial, sans-serif" font-size="${size * 0.4}"
        fill="white" text-anchor="middle" dominant-baseline="middle" dy=".3em">${text}</text>
</svg>
  `.trim();
  return svg;
}

function createPNGFromSVG(svg, outputPath) {
  // For now, save as SVG (browsers can convert)
  // In production, you'd use sharp or canvas library
  const pngPath = outputPath.replace('.svg', '.png');

  // Use a simple approach: create a minimal valid PNG
  // This is a 1x1 blue PNG (base64 encoded)
  const minimalPNG = Buffer.from(
    'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==',
    'base64'
  );

  // Copy the minimal PNG (we'll replace with proper icons later)
  fs.writeFileSync(pngPath, minimalPNG);
  console.log(`✅ Created ${pngPath}`);

  // Also save SVG for reference
  fs.writeFileSync(outputPath, svg);
  console.log(`✅ Created ${outputPath}`);
}

function main() {
  const iconsDir = path.join(__dirname, '..', 'public', 'icons');
  fs.mkdirSync(iconsDir, { recursive: true });

  // Create 192x192 icon
  const svg192 = createSVGIcon(192, 'E2T');
  createPNGFromSVG(svg192, path.join(iconsDir, 'icon-192x192.svg'));

  // Create 512x512 icon
  const svg512 = createSVGIcon(512, 'E2T');
  createPNGFromSVG(svg512, path.join(iconsDir, 'icon-512x512.svg'));

  console.log('✅ PWA icons created successfully');
  console.log('⚠️  Note: Using placeholder icons. Replace with proper PNG files for production.');
}

main();

File: frontend/scripts/create-pwa-icons.py
#!/usr/bin/env python3
"""Create simple PWA icons using PIL"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size, text, output_path):
    """Create a simple square icon with text"""
    # Create image
    img = Image.new('RGB', (size, size), color='#0f172a')

    # Draw text
    draw = ImageDraw.Draw(img)

    # Try to use a nice font, fallback to default
    try:
        font_size = int(size * 0.4)
        font = ImageFont.truetype('/System/Library/Fonts/Helvetica.ttc', font_size)
    except:
        font = ImageFont.load_default()

    # Get text bounding box
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]

    # Center text
    x = (size - text_width) / 2
    y = (size - text_height) / 2

    draw.text((x, y), text, fill='white', font=font)

    # Save
    img.save(output_path)
    print(f"✅ Created {output_path} ({size}x{size})")

def main():
    os.makedirs('public/icons', exist_ok=True)

    # Create icons
    create_icon(192, 'E2T', 'public/icons/icon-192x192.png')
    create_icon(512, 'E2T', 'public/icons/icon-512x512.png')

    print("✅ PWA icons created successfully")

if __name__ == '__main__':
    main()
